libs/vkd3d/command.c

```diff
@@ -2578,65 +2578,41 @@ static void d3d12_command_list_clear_attachment_pass(struct d3d12_command_list *
         if (clear_aspects & VK_IMAGE_ASPECT_STENCIL_BIT)
             plane_write_mask |= VKD3D_STENCIL_PLANE_OPTIMAL;
 
-        attachment_ref.layout = VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL;
-        attachment_desc.finalLayout = dsv_plane_optimal_mask_to_layout(
+        final_layouts[0] = dsv_plane_optimal_mask_to_layout(
                 d3d12_command_list_notify_dsv_writes(list, resource, view, plane_write_mask),
                 resource->format->vk_aspect_mask);
 
         if (separate_ds_layouts)
         {
-            stencil_attachment_desc.stencilFinalLayout = vk_separate_stencil_layout(attachment_desc.finalLayout);
-            attachment_desc.finalLayout = vk_separate_depth_layout(attachment_desc.finalLayout);
+            /* Do not transition aspects that we are not supposed to clear */
+            attachment_info.imageLayout = (clear_aspects & VK_IMAGE_ASPECT_DEPTH_BIT)
```

Comment:
Wouldn't it be more correct to use attachment desc layout = final_layout here? The clear aspects we pass in to notify_dsv_writes will promote anything to OPTIMAL if we have writes.

Reply:
as in:
```
            /* Do not transition aspects that we are not supposed to clear */
            final_layouts[1] = vk_separate_stencil_layout(final_layouts[0]);
            final_layouts[0] = vk_separate_depth_layout(final_layouts[0]);

            attachment_info.imageLayout = final_layouts[0];
            stencil_attachment_info.imageLayout = final_layouts[1];
```
?

Reply:
Yes.